src/types.cpp

```diff
@@ -561,13 +561,14 @@ std::weak_ptr<const Struct> SizedType::GetStruct() const
 
 bool SizedType::FitsInto(const SizedType &t) const
 {
-  if (IsStringTy() && t.IsStringTy())
-    return GetSize() <= t.GetSize();
```

Comment:
This logic looks pretty sensible - why get rid of it?

Reply:
Nothing invokes this call path.

Reply:
This script does go through it, but passes the check: `BEGIN { $a = ("a", 1); $a = ("abc", 1);}`

Maybe there's no script that can fail the check though? Still might be nice to keep as a safety net

Reply:
Ah right, it happens for the inner tuple case. I think I got rid of it because of the type manipulation on both sides. Either way, I'll put it back because I'm not going to do that.